### PR TITLE
M3-3563 image events

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Phase 2 of getting progress bars displayed when creating an Image. This adds a progress event to images on Images Landing. Next up is adding create/update image requests to Redux.

## Type of Change

- Non breaking change ('update', 'change')

## Note to Reviewers

Create an image, then reload the page. Look in devtools and you should see an event with a secondary entity attached to the currently-being-created Image.

### CODE REVIEW

File: packages/manager/src/features/Images/ImagesLanding.tsx [hunk 1]
```diff
@@ -476,18 +473,41 @@ const EmptyCopy = () => (
...
+    const events = imageEvents(state.events);
+    const privateImagesWithEvents = Object.values(data).reduce(
+      (accum, thisImage) =>
+        produce(accum, draft => {
+          if (!thisImage.is_public) {
+            // The imageEvents selector above will filter out anything without a secondary_entity, so this is safe (I promise!)
+            // NB: the secondary_entity returns only the numeric portion of the image ID so we have to interpolate.
+            const matchingEvent = events.find(
+              thisEvent =>
+                `private/${thisEvent.secondary_entity!.id}` === thisImage.id
```
Comment (johnwcallahan) [inline packages/manager/src/features/Images/ImagesLanding.tsx]: I think we should still do:

```typescript
const matchingEvent = events.find(
  thisEvent =>
    thisEvent.secondary_entity &&
    `private/${thisEvent.secondary_entity!.id}` === thisImage.id
);
```

_Just in case_ we ever change the `imageEvents()` selector or something else.

Comment (Jskobos) [reply packages/manager/src/features/Images/ImagesLanding.tsx]: Fair enough

File: packages/manager/src/store/selectors/imageEvents.ts [hunk 2]
```diff
@@ -9,12 +9,13 @@ import { isInProgressEvent } from 'src/store/events/event.helpers';
...
-      !!thisEvent.secondary_entity &&
-      isInProgressEvent(thisEvent)
-  );
-};
+export default createSelector<ApplicationState['events'], Event[], Event[]>(
+  state => state.events || [],
+  events =>
+    events.filter(
+      (thisEvent: Event) =>
+        thisEvent.action === 'disk_imagize' &&
```
Comment (martinmckenna) [inline packages/manager/src/store/selectors/imageEvents.ts]: looks like `linode_create` also has an image as second entity.

**Comment by martinmckenna** [comment]
Unrelated: this button does not pre-fill the create flow with the image:

Screen Shot 2019-11-22 at 10 51 14 AM

File: packages/manager/src/store/selectors/imageEvents.ts (hunk 2 above)
Comment (Jskobos) [reply packages/manager/src/store/selectors/imageEvents.ts]: I'm not sure we should include that in this selector, though. `linode_create` has `progress_complete`, so if you e.g. deployed a Linode from one of your private images, then navigated to /images when the deployment was in process, you'd see the Image you're using with a progress bar, even though nothing's happening to the Image itself. We could filter those out in the consumer, but that sort of undercuts the rationale for having a selector.

Maybe we could rename this selector to make it more clear what it's being used for.

File: packages/manager/src/features/Images/ImagesLanding.tsx (hunk 1 above)
Comment (acourdavault) [reply packages/manager/src/features/Images/ImagesLanding.tsx]: On this My linter complains about the double check for `secondary_entity` being `null`
We could simplify to this: `thisEvent.secondary_entity!.id` => `thisEvent.secondary_entity.id`

Looks good

**Pull request merged**